text/0122-sdk-hub-scope-merge.md

```diff
@@ -0,0 +1,305 @@
+
+## Isolation Scopes
+
+Furthermore, there can also be **Isolation Scopes**.
+Similar to the global scope, these are also applied to events. However, isolation scopes can be created, either by us internally (the most common scenario), or also by users. The new APIs for this are:
+
+```js
+// Returns the currently active isolation scope.
+export function getIsolationScope(): Scope;
+
+// Create a new isolation scope for this scope
+// This will NOT make this scope the isolation scope, but will create a new isolation scope (based on the currently active isolation scope, if one exists)
+scope.isolate();
+
+// Similar to `withScope`, but it forks a new scope AND sets a new isolation scope for this context
+export function withIsolationScope(callback: (scope) => void): void;
+```
+
+You can fetch the currently active isolation scope via `getIsolationScope()`. You can define a new isolation scope via `scope.isolate()`, which will define a new isolation scope for this scope, and for all scopes that will be forked off this scope. When a client is created & bound, an initial isolation scope will immediately be created, similar to the global scope for a client.
+
+An isolation scope is attached to the current execution context, similar to the active scope. There is always exactly one active isolation scope. If you call `getIsolationScope()` before a client has been created, a noop isolation scope is returned, which should be merged in once a client is actually created (same as with the global scope).
+
+Similar to the global scope, an isolation scope is always a separate scope, so nothing will inherit off it - except for a potential superseding isolation scope. 
+If an isolation scope is created, and there is already an isolation scope in the current execution context, then the new isolation scope should be forked off the previous one (with copy-on-write).
+
+### When to create an isolation scope
+
+For most server-side SDKs, an isolation scope will be created for each request being processed. 
+Roughly, it will equate to each time we currently fork a hub.
+
+### Examples for isolation scopes
+
+Example for instrumentation that we would write:
+
+```ts
+function wrapHttpServerRequest(original: Function): Function {
+  // Fork an execution context for this server request, that is isolated
+  return Sentry.withIsolatedScope((scope) => {
+    // anything in here will have the same isolated scope!
+    return original();
+  })
+}
+```
+
+Example for hooking into external auto-instrumentation (e.g. OpenTelemetry):
+
+```ts
+let onRequestHook: (span: Span) => void;
+
+// This method is not defined by us, but is some external code
+// Here just for demonstration purposes of how that may be implemented
+function otelWrapHttpServerRequest(original: Function): Function {
+  // Fork an execution context for this server request,
```

Comment:
what is the semantic we are trying to achieve with isolate? is it per execution context? then why are we using 'fork an execution context' here. sorry but it really isn't clear to me what we're trying to achieve with this example. What kind of data lives on the `withScope` and what happens after you do `scope.isolate`?